diff --git a/hack/config-go.sh b/hack/config-go.sh
index 1023b82..1e84144 100644
--- a/hack/config-go.sh
+++ b/hack/config-go.sh
@@ -59,7 +59,8 @@ mkdir -p "${KUBE_GO_PACKAGE_BASEDIR}"
 # Create symlink under output/go/src.
 ln -snf "${KUBE_REPO_ROOT}" "${KUBE_GO_PACKAGE_DIR}"
 
-GOPATH="${KUBE_TARGET}:${KUBE_REPO_ROOT}/third_party"
+ORIG_GOPATH="$GOPATH"
+GOPATH="${KUBE_TARGET}:${KUBE_REPO_ROOT}/third_party${ORIG_GOPATH:+:$ORIG_GOPATH}"
 export GOPATH
 
 # Unset GOBIN in case it already exsits in the current session.
diff --git a/hack/install-package.sh b/hack/install-package.sh
new file mode 100755
index 0000000..89acb33
--- /dev/null
+++ b/hack/install-package.sh	2014-08-05 10:19:30.348810953 -0400
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script sets up a go workspace locally and builds all go components.
+
+set -e
+
+source $(dirname $0)/config-go.sh
+
+cd "${KUBE_TARGET}"
+
+if [[ $# == 0 ]]; then
+  # Update $@ with the default list of daemons needing renamed.
+  set -- proxy apiserver controller-manager kubelet kubecfg
+fi
+
+install -d $DESTDIR/usr/bin
+echo "+++ INSTALLING kubelet"
+install -p -m 755 bin/kubelet $DESTDIR/usr/bin/kubelet
+echo "+++ INSTALLING kubecfg"
+install -p -m 755 bin/kubecfg $DESTDIR/usr/bin/kubecfg
+for arg; do
+  echo "+++ INSTALLING ${arg}"
+  install -p -m 755 bin/${arg} $DESTDIR/usr/bin/kube-${arg}
+done
