From 2e69f978ca14913ca0f9f10c255d1bfc1c7cff59 Mon Sep 17 00:00:00 2001
From: Eric Paris <eparis@redhat.com>
Date: Wed, 13 Aug 2014 12:50:48 -0400
Subject: [PATCH 2/3] Respect version passed in via environment variable

---
 hack/version-gen.sh | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/hack/version-gen.sh b/hack/version-gen.sh
index 77082a8..61a856a 100755
--- a/hack/version-gen.sh
+++ b/hack/version-gen.sh
@@ -21,8 +21,11 @@ set -o pipefail
 topdir=$(dirname "$0")/..
 cd "${topdir}"
 
+if [ ! -z ${git_commit-} ]; then
+  git_commit=${git_commit}
+  # just use what was passed in....
+elif git_commit=$(git rev-parse --short "HEAD^{commit}" 2>/dev/null); then
 # TODO: when we start making tags, switch to git describe?
-if git_commit=$(git rev-parse --short "HEAD^{commit}" 2>/dev/null); then
   # Remove any invalid characters that might confuse "sed".
   git_commit=${git_commit//[^a-f0-9]/}
 
-- 
2.0.4

